natero/lib/natero/account.rb

```diff
@@ -1,144 +1,69 @@
-class Natero::Account
-  include HTTParty
-  include Serializable
-
-  BASE_URI = 'https://api.natero.com'
-  VERSION_URI = '/api/v2'
-
-  attr_reader :account_id, :name, :join_date, :renewal_date,
-              :billing_account_id, :support_account_id, :crm_account_id,
-              :billing_street, :billing_city, :billing_postal_code,
-              :billing_state, :billing_country, :phone, :employees,
-              :industry, :tier, :csm_score, :sales_rep_name, :sales_rep_email,
-              :source, :stage, :is_deleted, :is_churned, :inactive_time, :inactive_reason,
-              :assigned_csms, :custom_label_dimensions, :custom_value_dimensions,
-              :custom_event_dimensions, :stage_history, :nps_history, :raw_response
-
-  # http://apidocs.natero.com/apidoc.html#Accounts_Retrieve%20all%20accounts
+class Natero::Account < Natero::Base
   def self.retrieve_all
-    Natero::Response.new(get(endpoint))
+    get
   end
 
-  # http://apidocs.natero.com/apidoc.html#Accounts_Retrieve%20an%20account
   def self.retrieve(id)
-    Natero::Response.new(get(endpoint(id)))
+    get([id])
   end
 
-  # http://apidocs.natero.com/apidoc.html#Accounts_Bulk%20insert%2Fmodify%20accounts
   def self.bulk_insert_modify(accounts)
-    body = Natero.to_records_json(accounts)
-    Natero::Response.new(post(endpoint, { :body => body, :headers => { 'Content-Type' => 'application/json' } }))
-  end
+    post([], json_data(Natero.to_records_json(accounts)))
+  ende
 
-  # http://apidocs.natero.com/apidoc.html#Accounts_Modify%20an%20account
   def self.modify(id, account)
```

Comment:
Is changing the interface an option? Seems weird to leave all these as class methods and having to pass in the `id` every time instead of doing having an object that you interact with that knows its own id from initialization